### ISSUE TITLE
OPAL_CHECK_COMPILER_VERSION warnings

### ISSUE THREAD

**Issue by rhc54** [open-mpi/ompi#9404]
With current head of master:

```
configure.ac:600: the top level
configure.ac:601: warning: The macro `AC_TRY_RUN' is obsolete.
configure.ac:601: You should run autoupdate.
./lib/autoconf/general.m4:2997: AC_TRY_RUN is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
config/opal_check_compiler_version.m4:28: OPAL_CHECK_COMPILER is expanded from...
config/opal_check_compiler_version.m4:19: OPAL_CHECK_COMPILER_VERSION_ID is expanded from...
```

**Comment by rhc54** [comment]
Looks like someone introduced some changes that are not friendly to autoconf 2.7x as there are a number of warnings now coming out of master like:
```
configure.ac:945: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:945: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
confdb/aclocal_cc.m4:1223: PAC_FUNC_NEEDS_DECL is expanded from...
configure.ac:945: the top level
```
We had this nicely cleaned up - perhaps we need a way to ensure it doesn't get messed up again?

**Comment by jsquyres** [comment]
I thought the AC 2.7x updates we did simply fixed the *errors*, but not necessarily the *warnings*...?

**Comment by rhc54** [comment]
No, a couple of us worked on it and fixed the warnings too, except for romio since we didn't control it. These are coming from OMPI's own configure.

**Comment by jsquyres** [comment]
Yes, I don't think we ever fully fixed *all* the warnings in Open MPI's `configure`.  For example, I see https://github.com/open-mpi/ompi/blame/3fedbf777c4864c9c4f864228d787b27fadf0f74/config/opal_setup_cc.m4#L335, which shows that the `AC_TRY_LINK` on that line -- which emits a warning in AC 2.7x -- was put there 13 years ago and hasn't been edited since.

That being said, I definitely agree that we should clean these up when/where people have time to do so.

**Comment by rhc54** [comment]
> when/where people have time to do so.

Which means it will never happen. I don't think people realize that for those of us who aren't _required_ to work on OMPI, wading thru a tsunami of warnings every time we want to touch the code is a major disincentive - just smacks of a pretty poor code base. OMPI literally dumps out _pages_ of warnings in both configure and compile, regardless of which system I use.

Pretty pathetic 😞

**Comment by jsquyres** [comment]
I believe that this has been fixed in master and v5.0.x (disclaimer: the errors that are left art part of ROMIO).
